invenio_drafts_resources/auditlog_actions.py

```diff
@@ -8,30 +8,119 @@
 
 """Action registration via entrypoint function."""
 
-from invenio_audit_logs.actions import AuditAction
-
-
-def record_actions():
-    """Function to add actions to the registry."""
-    return {
-        "draft.create": AuditAction(
-            name="draft.create",
-            message_template="User {user_id} created the draft {resource_id}.",
-        ),
-        "draft.edit": AuditAction(
-            name="draft.edit",
-            message_template="User {user_id} updated the draft {resource_id}.",
-        ),
-        "record.publish": AuditAction(
-            name="record.publish",
-            message_template="User {user_id} published the record {resource_id}.",
-        ),
-        "draft.delete": AuditAction(
-            name="draft.delete",
-            message_template="User {user_id} deleted the draft {resource_id}.",
-        ),
-        "draft.new_version": AuditAction(
-            name="draft.new_version",
-            message_template="User {user_id} created a new version {resource_id}.",
-        ),
-    }
+from invenio_access.permissions import system_user_id
+from invenio_audit_logs.services import AuditLogAction
+from invenio_records.dictutils import dict_lookup, dict_set
+from invenio_users_resources.entity_resolvers import UserResolver
+
+
+class UserContext:
+    """Payload generator for audit log using the user entity resolver."""
+
+    def __init__(self, key="user"):
+        """Ctor."""
+        self.key = key
+        self.resolver = UserResolver()
+
+    def __call__(self, data, lookup_key="user_id", **kwargs):
+        """Update data with resolved user data."""
+        entity_ref = dict_lookup(data, lookup_key)
+        entity_proxy = self.resolver.get_entity_proxy({self.key: entity_ref})
+        if entity_ref == system_user_id:
+            entity_data = entity_proxy.system_record()
+            entity_data = {
```

Comment:
We move the picking of fields needed for audit logs to audit logs, to keep this generic